shared/reducers/chat.js

```diff
@@ -278,25 +288,25 @@ function reducer(state: Types.State = initialState, action: ChatGen.Actions) {
     case ChatGen.exitSearch: {
       return state.set('inSearch', false)
     }
-    case 'teams:setChannelCreationError': {
-      const {payload: {channelCreationError}} = action
-      return state.set('channelCreationError', channelCreationError)
+    case TeamsGen.setChannelCreationError: {
+      const {error} = action.payload
+      return state.set('channelCreationError', error)
     }
-    case 'teams:setTeamCreationError': {
-      const {payload: {teamCreationError}} = action
-      return state.set('teamCreationError', teamCreationError)
+    case TeamsGen.setTeamCreationError: {
+      const {error} = action.payload
+      return state.set('teamCreationError', error)
     }
-    case 'teams:setTeamCreationPending': {
-      const {payload: {teamCreationPending}} = action
-      return state.set('teamCreationPending', teamCreationPending)
+    case TeamsGen.setTeamCreationPending: {
+      const {pending} = action.payload
+      return state.set('teamCreationPending', pending)
     }
-    case 'teams:setTeamJoinError': {
-      const {payload: {teamJoinError}} = action
-      return state.set('teamJoinError', teamJoinError)
+    case TeamsGen.setTeamJoinError: {
+      const {error} = action.payload
+      return state.set('teamJoinError', error)
     }
-    case 'teams:setTeamJoinSuccess': {
-      const {payload: {teamJoinSuccess, teamname}} = action
-      return state.set('teamJoinSuccess', teamJoinSuccess).set('teamJoinSuccessTeamName', teamname)
+    case TeamsGen.setTeamJoinSuccess: {
+      const {success, teamname} = action.payload
```

Comment:
this form is used more in our codebase and i also think nested destructuring on the left side is less readable. This is just one level deep so its not that confusing but things like
```
const {foo: {bar: {baz: username}, other: somethingElse}} = store
```
is harder to read than
```
const username = store.foo.bar.baz
const somethingElse = store.foobar.other // or whatever
```